README.md
# p5-Bb-Collaborate
Perl bindings for BlackBoard Collaborate virtual classroom software

Folders:

- Elive/ - Original ELM Bindings, now base class for Bb::Collaborate::V3
- Bb-Collaborate-V3/ - Collaborate SAS V3 bindings

See individual README's in above folders.


Elive/Changes
Revision history for Elive

1.36   1st November 2015
       [Elive::DAO]
       - added support for class level invocation of Elive::DAO update and delete methods.
       Note: that all primary keys and mandatory fields need to be supplied.
       [distribution]
       - moved source repository to https://github.com/dwarring/p5-Bb-Collaborate.git

1.35   7th June 2015
       Support for new module Bb::Collaborate::V3.
       [Elive::DAO]
       - added list() params options
       [Elive::Util::Type]
       - removed Elive:: namespace dependency.
       [scripts/elive_query]
       - added --adapter=v3 option (Bb::Collaborate::V3). removed deprecated v2 adapter

1.34   4th April 2015
       - Update copyright notices
       [Elive::DAO]
       - tweaks for latest (SAS compatibile) Elive::StandardV3
       [kwalitee]
       - Use consistant version number in sub-classes

1.33   22nd January 2015
       [bug-fixes]
       - RT#101018 "Mouse 2.4.1 when Test::Script::Run is installed"
         Removed fragile 'stderr empty' tests from t/script-elive_query.t
         and t/script-elive_raise_meeting.t.
       - fixed a taint issue in Elive::Entity::Preload.

1.32   1st March 2014
       [testing]
       - raised max tested ELM version 11.1.2 => 12.5.4
       - fixed soap-session-connect.t test plan
       [distribution]
       - removed Elive::Array
       - relocated sourceforge repository

1.31   28th March 2013
       [testing]
       - added Test::More 0.98 dependency. Older versions don't understand
         $t->is_passing

1.30   4th January 2013
       [bug fixes]
       Elive::DAO
       - Relax HiResDate constraints to allow negative timestamps; downgrade
         to a warning

1.29   25th October 2012
       [enhancements]

       Elive::Entity::Meeting, Elive::Entity::Session, Elive::View::Meeting
       - buildJNLP() method: Tidy up of POD and argument handling.
         -- removed `version' argument; not used.
         -- added 'sessionRole' parameter.
         -- accept any one of 'userId', 'userName' or 'displayName'. In
            particular, don't default 'userId' when 'displayName' given.

       Elive::Entity::Role - added some constants:
       - ${Elive::Entity::Role::PARTICIPANT}       (3)
       - ${Elive::Entity::Role::MODERATOR}         (2)
       - ${Elive::Entity::Role::APPLICATION_ADMIN} (1)
       - ${Elive::Entity::Role::SYSTEM_ADMIN}      (0)

       Elive::Entity::Participant - added some constants:
       - ${Elive::Entity::Participant::TYPE_USER}  (0)
       - ${Elive::Entity::Participant::TYPE_GROUP} (1)
       - ${Elive::Entity::Participant::TYPE_GUEST} (2)

       Elive::DAO
       - revert() method - Silence 'unsaved changes' warnings

1.28   12th October 2012
       [bug fixes]
       elive_raise_meeting, script-elive_raise_meeting.t
       - fixed -moderators option. These were being downgraded to regular
         participants. Added tests (bug introduced in 1.20)
       Elive::Entity::InvitedGuest, soap-session-participants.t
       - added BUILDARGS method plus string coercement (it's never had these).
         The following now works:
                Elive::Entity::InvitedGuest->new('Robert (bob@example.com)');
       [enhancements]
       Elive:
       - server_details() - Return all server(s) in a list context:
                my @servers = Elive->server_details;

1.27   14th July 2012
       [bug fixes]
       Elive::Entity::Participant
       - tightened up handling of reference structures (BUILDARGS method)
       - fixed bug with handling of invited guest objects.
       Elive::Entity::Group, Elive::Util::Type, dao-construct.t
       - Fixed element_class declaration and tightened tests to match actual
         content. Elements may be subgroups (group objects) or user-ids
         (strings).
       [testing]
       - dao-errors.t - now checks content of error messages
       - script-elive_lint_conf.t - tolerate perl 5.17.x depreciation warnings 
         (from XML::Simple <= 2.20).
       - soap-preload.t - cross-check imported preload sizes

1.26   4th May 2012
       [bug fixes]
       Elive::Entity::ParticipantList, elive_raise_meeting:
       - Fixed to align with uid (loginName) case sensitivity policies under
         LDAP and latest (ELM 11.1.2). The case of the uid is preserved on
         entry. But comparisons and uniqueness checks are case insensitive.
         -- E.g. if a user has an LDAP uid of 'johnsmith'.  Attempting to add
            'JohnSmith' now works.
         -- Note that after update, the returned participant list entry 
            contains the response uid, e.g. 'johnsmith'.

1.25   18th April 2012
       [enhancements]
       Elive::Connection
       - Added timeout option. This is passed through to the SOAP::Lite proxy object.
       [testing]
       - soap-participants.t, soap-session-participants.t - stress tests now passing.

1.24   28th February 2012
       [Tested against Blackboard Elive 11.1.2  / ELM 3.5.0.]
       [testing]
       - soap-connect.t - raised max_version to 11.1.2
       - soap-participant-list.t - long participant / stress-tests are hanging
         under LDAP. I marked these tests as TODO and added an alarm-timeout.
       [bug fixes]
       elive_query
        - removed disused _get_user() function.
        - direct 'connection error ...' message to STDERR

1.23    26th January 2012
        [enhancements]
        Elive::Entity
        - added quote() method - for quoting list() filter arguments.
        [bug fixes]
        - elive_raise_meeting: correctly handle login and group names with
          embedded single quotes, using Elive::Entity->quote().

1.22    5th January 2012
        [security]
        Elive::Util::_tidy_decimal():
        - Untaint Ints once sanitised - for consistency with Enums and Bools.
        [testing]
        - Replaced Test::Builder->new with Test::More->builder
        00load.t, soap-connect.t:
        - now bail testing if any sub tests fail

Elive/Makefile.PL
# -*- perl -*-

use strict;
use warnings;
use inc::Module::Install;

name           ('Elive');
module_name    ('Elive');
author         ('David Warring <david.warring@gmail.com>');
all_from       ('lib/Elive.pm');
license	       ('perl');
install_script ('elive_query', 'elive_raise_meeting', 'elive_lint_config');

requires(
    'Class::Accessor'              => 0,
    'Class::Data::Inheritable'     => 0,
    'Clone'                        => 0,
    'Date::Parse'                  => 0,
    'Carp'                         => 0,  # for scripts
    'File::Spec'                   => 0,
    'File::Basename'               => 0,
    'Getopt::Long'                 => 2.35,  # for scripts
    'HTML::Entities'               => 0,
    'IO::Interactive'              => 0,
    'MIME::Types'                  => 0,
    'Mouse'                        => 0.81,
    'Mouse::Util::TypeConstraints' => 0,
    'Pod::Usage'                   => 1.33,  # for elive_raise_meeting
    'parent'                       => 0,
    'Scalar::Util'                 => 0,
    'SOAP::Lite'                   => 0.712,
    'Term::ReadKey'                => 0,
    'Term::ReadLine'               => 0,
    'Time::Local'                  => 0,
    'Try::Tiny'                    => 0,
    'URI'                          => 0,
    'URI::Escape'                  => 0,
    'YAML::Syck'                   => 0,
    'XML::Simple'                  => 0,
    );

recommends(
    'Term::ReadLine::Perl'         => 0,  
    );

test_requires (
    'File::Temp'                   => 0,
    'Test::More'                   => 0.98,
    'Test::Warn'                   => 0,
    'Test::Fatal'                  => 0,
    'version'                      => 0,
    );

#
# Module::Install::Repository magic
#
repository("git://git.code.sf.net/p/elive/git");

WriteAll;
